src/battle_ai_util.c

```diff
@@ -608,22 +609,22 @@ void SetBattlerData(u8 battlerId)
         }
 
         // Use the known battler's ability.
-        if (BATTLE_HISTORY->abilities[battlerId] != ABILITY_NONE)
-            gBattleMons[battlerId].ability = BATTLE_HISTORY->abilities[battlerId];
+        if (AI_PARTY->mons[side][battlerId].ability != ABILITY_NONE)
+            gBattleMons[battlerId].ability = AI_PARTY->mons[side][battlerId].ability;
```

Comment:
The `battlerId`s here are ones for Pokémon currently in battle (0-3), but `AI_PARTY` is about the opponent's full party.

Reply:
Should be AI_PARTY->mons[side][gBattlerPartyIndexes[battler]]

Reply:
Makes sense.

Applied changes and squashed.